ui/lib/kv/addon/templates/secret/metadata/index.hbs

```diff
@@ -4,8 +4,14 @@
 ~}}
 
 <Page::Secret::Metadata::Details
+  @backend={{this.model.backend}}
+  @breadcrumbs={{this.breadcrumbs}}
+  @canDeleteMetadata={{this.model.permissions.metadata.canDelete}}
+  @canReadCustomMetadata={{or this.model.permissions.metadata.canRead this.model.permissions.data.canRead}}
+  @canReadMetadata={{this.model.permissions.metadata.canRead}}
+  @canUpdateMetadata={{this.model.permissions.metadata.canUpdate}}
   @metadata={{this.model.metadata}}
   @path={{this.model.path}}
-  @breadcrumbs={{this.breadcrumbs}}
+  {{! only fetched if a user cannot read metadata }}
   @secret={{this.model.secret}}
```

Comment:
What if instead you passed in `@customMetadata`, and then you could eliminate handling the which case it is in the page component
```
@customMetadata={{or this.model.metadata.custom_metadata this.model.secret.custom_metadata}}
```